vpr/src/route/route_profiling.cpp

```diff
@@ -213,7 +213,9 @@ void profiling_initialization(unsigned max_fanout) {
 void conn_start() {
     conn_start_time = clock();
 }
+
 void conn_finish(int src_rr, int sink_rr, float criticality) {
+    auto& device_ctx = g_vpr_ctx.device();
```

Comment:
Should we use ``const auto&`` since you are not going to modify anything in device modeling.